### PR TITLE
feat: remove support for internet explorer

### PR DESCRIPTION
**Pull request by langz**
This PR removes support for Internet Explorer.

Changes:
- Removed the helpers, and usage of `isIE11` and `IS_IE11` from [packages/dnb-eufemia/src/shared/helpers.js](https://github.com/dnbexperience/eufemia/pull/1807/files#diff-c417123e364321f78abb041ebe1f757eca354b85bd37a2146dccbdf717234f26)
- Removed the 4th parameter(`$is1px: false /* IE11 fix */`) from `@mixin fakeBorder` of [packages/dnb-eufemia/src/style/core/utilities.scss](https://github.com/dnbexperience/eufemia/pull/1807/files#diff-eba6a70067334bebb35b89e63c1ffcde7f3a129206834b9e6283dfbb05f0a9af), and the usage of it.
- Edited [packages/dnb-design-system-portal/.browserslistrc](https://github.com/dnbexperience/eufemia/pull/1807/files#diff-be68f9a1b4a7aba0cb2a823ca0368389243ba8fc00d3b8605ebe6941ae82c171)
- Edited [packages/dnb-eufemia/.browserslistrc](https://github.com/dnbexperience/eufemia/pull/1807/files#diff-a460bf5e0739566160b80df8ee43398d1e7eea20262b1705909e4f9906413a5f)

Thoughts:
- Trying to be careful not removing all ms specific css properties like `-ms-*`, i.e `-ms-overflow-style`, as it's also used/supported by Edge

### CODE REVIEW

File: packages/dnb-eufemia/src/components/global-status/GlobalStatus.js [hunk 1]
```diff
@@ -501,11 +500,7 @@ export default class GlobalStatus extends React.PureComponent {
     try {
       const element = this._wrapperRef.current
       this._scrollToStatusTimeout = isElementVisible(element, isDone)
-      if (
-        element &&
-        !IS_IE11 &&
```
Comment (langz) [inline packages/dnb-eufemia/src/components/global-status/GlobalStatus.js:506]: Not 100% sure about this one 🤔

File: packages/dnb-eufemia/src/components/number-format/NumberUtils.js [hunk 2]
```diff
@@ -981,7 +972,6 @@ export function getFallbackCurrencyDisplay(
     !currency_display &&
     // Safari does not support `narrowSymbol` for now, so `symbol` will be used then.
     !IS_SAFARI &&
-    !IS_IE11 &&
```
Comment (langz) [inline packages/dnb-eufemia/src/components/number-format/NumberUtils.js:984]: Not 100% sure about this one 🤔

File: packages/dnb-eufemia/src/style/stylis.js [hunk 3]
```diff
@@ -6,13 +6,12 @@
 
 import { matchAll } from '../shared/component-helper'
 import properties from './properties'
-import { IS_IE11 } from '../shared/helpers'
 
 const findRegExp = /var\(([^)]*)\)/g
 
 const stylisPlugin = (context, content, props = null) => {
   // Only use the property context
-  if (context !== 1 || (props === null && !IS_IE11)) {
```
Comment (langz) [inline packages/dnb-eufemia/src/style/stylis.js:15]: Not 100% sure about this one 🤔

**Comment by langz** [comment]
We should perhaps do something with https://github.com/dnbexperience/eufemia/blob/bc3f12a24d3912ccbe3cf4593508c3e5ebd2c8d5/packages/dnb-eufemia/babel.config.js#L28 ?

File: packages/dnb-eufemia/src/components/global-status/GlobalStatus.js (hunk 1 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/global-status/GlobalStatus.js:506]: Seems fine to me

File: packages/dnb-eufemia/src/components/number-format/NumberUtils.js (hunk 2 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/number-format/NumberUtils.js:984]: Seems fine to me

File: packages/dnb-eufemia/src/style/stylis.js (hunk 3 above)
Comment (langz) [reply packages/dnb-eufemia/src/style/stylis.js:15]: Seems fine to me

File: packages/dnb-design-system-portal/src/docs/uilib/about-the-lib/releases/eufemia/v10-info.md [hunk 4]
```diff
@@ -67,6 +67,11 @@ v10 of @dnb/eufemia contains _breaking changes_. As a migration process, you can
 - `use_scrollwheel` and `on_init` properties, as well as the `raw_value` event value from [Slider](/uilib/components/slider) was removed in order to support multiple buttons.
 - Helper class `.dnb-sr-only--inline` and SCSS mixin `srOnlyInline` was removed.
 - Helper class `.dnb-not-sr-only` and SCSS mixin `notSrOnly` was removed.
+- Stopped supporting Internet Explorer (IE), as Microsoft formally ended support for IE in June, 2022.
+
+### [Stopped supporting Internet Explorer (IE)](/uilib/usage/#supported-browsers-and-platforms)
+
+- What does this mean for the users of Eufemia?
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/about-the-lib/releases/eufemia/v10-info.md:74]: TODO, write something informative

File: packages/dnb-design-system-portal/.browserslistrc [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # Should reflect: https://eufemia.dnb.no/uilib/usage#supported-browsers-and-platforms
-ie 11
 edge >= 14
 firefox >= 52
 chrome >= 49
-safari >= 10
\ No newline at end of file
+safari >= 10
+not ie > 0
+not ie_mob > 0
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/.browserslistrc:7]: Question: Should we change our strategy and do what the folks at browser list recommends?

> If you want to change the default set of browsers, we recommend including last 2 versions, not dead, > 0.2%. 

https://github.com/browserslist/browserslist#best-practices

File: packages/dnb-eufemia/scripts/prebuild/config/postcssConfig.js [hunk 6]
```diff
@@ -27,35 +27,6 @@ module.exports = ({ IE11 = false, sass = sassBin, ...options } = {}) => {
...
-    //       // importFrom: ['path/to/file.css'],
-    //       ...options
-    //     })
-    //   : null,
-    // require('autoprefixer')({
-    //   browsers: ['last 2 versions', IE11 ? 'explorer >= 11' : null].filter(
-    //     i => i
-    //   )
-    // })
   ].filter((i) => i)
```
Comment (tujoworker) [inline packages/dnb-eufemia/scripts/prebuild/config/postcssConfig.js:30]: I think, we even can remove more here. I made a commit with what I think we can remove.

**Review by tujoworker — APPROVED** [review]
Very nice work! 🙌💯

I had a look locally as well and have some commits I could push.

File: packages/dnb-design-system-portal/.browserslistrc (hunk 5 above)
Comment (tujoworker) [reply packages/dnb-design-system-portal/.browserslistrc:7]: hmm yes, I thought IE is dead. But it seems that the last one wins, so when I swap out the order, it looks better:

`last 2 versions, > 0.2%, not dead` https://browsersl.ist/#q=last+2+versions%2C+%3E+0.2%25%2C+not+dead

Comment (tujoworker) [reply packages/dnb-design-system-portal/.browserslistrc:7]: Maybe we should link to this page from our docs?

**Comment by tujoworker** [comment]
We should ensure that this PR gets in as a feature, so it gets mentioned in the change log:

> feat: remove support for internet explorer

When it comes to the [browserlist updates](https://github.com/dnbexperience/eufemia/pull/1807#discussion_r1049545439), we should do that in another PR, because it effects "other browsers".

**Comment by langz** [comment]
> We should ensure that this PR gets in as a feature, so it gets mentioned in the change log:
> 
> > feat: remove support for internet explorer
> 
> When it comes to the [browserlist updates](https://github.com/dnbexperience/eufemia/pull/1807#discussion_r1049545439), we should do that in another PR, because it effects "other browsers".

I'll remove the browserlist updates for now then 🤝

File: packages/dnb-design-system-portal/.browserslistrc (hunk 5 above)
Comment (langz) [reply packages/dnb-design-system-portal/.browserslistrc:7]: I've added a few sentence and a link to this page in the following commit:
https://github.com/dnbexperience/eufemia/pull/1807/commits/e3a1f8ec02938829afa8c3b3280ba30f5ff3c0fc

**Pull request merged**